scripts/check_hook_registration.py

```
@@ -83,12 +103,170 @@ def check(name, condition):
+                  gate_exit_code_raw("git commit -n", temp_dir) == 2)
+            check("malformed payload without a commit passes",
+                  gate_exit_code_raw("just some text", temp_dir) == 0)
+            check("non-string command value never crashes the gate",
+                  gate_exit_code_raw('{"tool_input":{"command":null}}',
+                                     temp_dir) == 0)
+            # Missing git (empty PATH) must fail CLOSED, not crash open.
+            no_git_environment = dict(os.environ)
+            no_git_environment["GIT_CONFIG_GLOBAL"] = os.devnull
+            no_git_environment["GIT_CONFIG_SYSTEM"] = os.devnull
+            no_git_environment["PATH"] = ""
+            no_git = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)